pandas/io/formats/style.py

```diff
@@ -2171,9 +2206,12 @@ def _maybe_wrap_formatter(
         raise TypeError(f"'formatter' expected str or callable, got {type(formatter)}")
 
     if na_rep is None:
-        return formatter_func
+        return (lambda x: formatter_func(escape_func(x))) if escape else formatter_func
     else:
-        return lambda x: na_rep if pd.isna(x) else formatter_func(x)
+        if escape:
+            return lambda x: na_rep if pd.isna(x) else formatter_func(escape_func(x))
```

Comment:
This might be a bug escaping floats, since escaping converts to string. Will review and come back

Reply:
OK this is fixed now too. 3rd test added.